Prox/Prox/Utilities/PlaceUtilities.swift

```diff
@@ -21,6 +22,43 @@ struct PlaceUtilities {
         }
     }
 
+    static func sort(places: [Place], byTravelTimeFromLocation location: CLLocation, ascending: Bool = true, completion: @escaping ([Place]) -> ()) {
+        PlaceUtilities.getTravelTimes(forPlaces: places, fromLocation: location).upon { result in
+            let sortedPlaces = places.sorted { (placeA, placeB) -> Bool in
+                guard let placeATravelTimes = PlaceUtilities.lastTravelTimes(forPlace: placeA),
+                    let placeBTravelTimes = PlaceUtilities.lastTravelTimes(forPlace: placeB) else {
+                        return false
+                }
+
+                let placeAETA = placeATravelTimes.getShortestTravelTime()
+                let placeBETA = placeBTravelTimes.getShortestTravelTime()
+
+                guard ascending else {
+                    return placeAETA > placeBETA
+                }
+                return placeAETA < placeBETA
+            }
+            completion(sortedPlaces)
+        }
+    }
+
+    private static func getTravelTimes(forPlaces places: [Place], fromLocation location: CLLocation) -> Future<[DatabaseResult<TravelTimes>]> {
```

Comment:
nit: Not necessary to change and I'm not sure if it's your style, but you could replace this method with:
```swift
let placesWithTimes = places.map { place in place.travelTimes(fromLocation: location }
return placesWithAllTimes.allFilled()
```

which I find easier to quickly understand the intent (since a for loop can do anything but `map` has a specific result).